packages/flamegrill/src/profile/profile.ts

```diff
@@ -82,7 +93,7 @@ export async function profile(scenarios: Scenarios, config: Required<ScenarioCon
  * @param {string} logDir Absolute path to output log profiles.
  * @returns {string} Log file path associated with test.
  */
-async function profileUrl(browser: Browser, testUrl: string, profileName: string, logDir: string): Promise<Profile> {
+async function profileUrl(browser: Browser, testUrl: string, profileName: string, logDir: string, executeBeforeMeasurement?: (page: ProfilePage) => Promise<void>): Promise<Profile> {
```

Comment:
I think `onPageLoaded` might be a better term. There should be a `@param` comment added describing its use as an optional mechanism for doing page operations for measurement.

Reply:
AH thank you. i didn't like the name i had here but i couldn't think better

Reply:
actually, `onPageLoad` won't work if we support other options for `page.goto(url)`

Reply:
see: https://github.com/puppeteer/puppeteer/blob/master/docs/api.md#pagegotourl-options `waitUtil`

Reply:
maybe `onPageNavigated` ?

Reply:
I'm only talking about the name of the argument here..